### DISCUSSION TITLE
Windows Terminal v1.12.1073

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#12761] [Announcements]
This is a significant servicing release to fix a number of huge issues in 1.12.

This release was made available to insiders in the Dev and Beta channels on March 25, and will be generally available to
everyone once it is considered stable. You can always install the update from this release page or using `winget`.

<details>
<summary>Preinstallation Kit info</summary>

A [preinstallation](https://docs.microsoft.com/en-us/windows/msix/desktop/deploy-preinstalled-apps) kit is available for system integrators and OEMs interested in prepackaging Windows Terminal with a Windows image. More information is available in the [DISM documentation on preinstallation](https://docs.microsoft.com/windows-hardware/manufacture/desktop/preinstall-apps-using-dism). Users who do not intend to preinstall Windows Terminal should continue using the _msixbundle_ distribution.

</details>

## Changes

This version of Windows Terminal is now distributed in two bundles, one of which works on Windows 10-11 and the
other of which only works on Windows 11. The Windows 11 version is much smaller because we no longer need to work around
a platform issue related to our dependencies.

If you intend on using Terminal as an unpackaged application--that is, extracting the `msix` file--we recommend that
you use the `Win10` bundle. You will need the [Visual C++ runtime redistributable](https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist).

In addition, if you install the _packaged_ version on either Windows 10 or Windows 11, it now depends on [the Visual C++ Universal Runtime Package](https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/cpp/libraries/c-runtime-packages-desktop-bridge?msclkid=3e1dd82aac8b11ec8f9dbd86d5979ca8#how-to-install-and-update-desktop-framework-packages).

Despite these distributions having different version numbers, they are built from the same code and there is no
functional difference between them.

If you install the Windows 10 verison on Windows 11, it will probably automatically upgrade itself to the Windows 11
version. It turns out that it is impossible to have two bundles with _the same version number_, so it has to be this
way.

### Usability

* Terminal can once again be configured as a startup application, and can be detected by tools like PowerToys (#12491)
* There was a puzzling "Element not found" error during settings loading; there is no longer such an error (#12687)
* Terminal will no longer mix up profiles when it is launched in response to a console application spawning (#12484)
* Formatted copy will now try harder to preserve Unicode charatcers in RTF (#12586) (thanks @ianjoneill!)
* We have replaced the word "Summon" with "Show/Hide" in the command palette for improved localization (#12603)
* Our confidence in the settings UI's _Save_ button has led to us no longer backing up the settings JSON file (#12652)
   * We won't be deleting the 61,000 backups we did leave on your hard drive, but what's a couple thousand kilobytes between friends?

### Accessibility

* Terminal now announces newly-printed text to any attached screen reader (#12358)
* Command palette search now tries to announce the number of results to the screen reader (#12429)

### Reliability

* We won't crash any longer if you give us a command line that is a directory (#12538) (thanks @ianjoneill!)
* A crash on launch related to multi-windowing and the default terminal setting has been quashed (subset of #12205)
* Fixed a crash setting the hotkey during teardown (#12580)
* Fixed a different pair of crashes, also likely related to default terminal handoff (#12666)
* `ScrollConsoleScreenBuffer` no longer takes the console upstate (#12669)
* Pressing <kbd>Page Up</kbd> or <kbd>Page Down</kbd> with an empty command palette, which seemed like a reasonable thing to do, was taught to _not_ crash the Terminal (#12528)

### Rendering

* Font axes/features once again work across a DPI change (#12492)

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.12.10732.0'>Windows Terminal v1.12.1073</a>.</em>

**Comment by melanchall** [comment]
Hello,

Is all OK with the version?

I have 1.12.10393.0 installed, trying to install new version I get this message:

![image](https://user-images.githubusercontent.com/26361185/160227310-9dab7e4d-4bdd-42b0-9c33-1c9924fa03b0.png)

1073**2** is obviously greater than 10393 so this looks strange. I suppose there is no **2** last digit in this new release version in fact so real version is 1.12.1073 and 1073 is indeed less than 10393.

Can you please check the version in the installer?

**Reply by liyuhang1997** [reply]
Can't reproduce

**Reply by DHowett** [reply]
Unfortunately, I can't reproduce this either. What version of Windows are you on?

The version number in the package _is_ `1.12.10732.0`, but in the release notes I have chosen to remove the last digit because it changes between the Windows 10 and Windows 11 builds.

You can force this package to install with the PowerShell Add-AppxPackage command:

```
Add-AppxPackage -ForceUpdateFromAnyVersion path\to\whatever.msixbundle
```

**Reply by melanchall** [reply]
@DHowett I'm on Win 10 20H2 (Build 19042.1586). PowerShell command gives me the error:

```
Add-AppxPackage : Deployment failed with HRESULT: 0x80073CF3, Package failed updates, dependency or conflict validation.
```

Can I just remove Windows Terminal and install latest version? All profiles and settings won't lost, right?

**Reply by juzzuf** [reply]
I get that exact same error window as in the first post when trying to upgrade from 1.12.10393.0 to 1.12.10732.0 on Windows 10 21H2 (OS Build 19044.1586).

**Reply by DHowett** [reply]
Very curious...

Can either of you run `Get-AppxPackage Microsoft.VCLibs.140.00.UWPDesktop` and report the output here?

**Reply by melanchall** [reply]
@DHowett output of the command:

```
Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X86
ResourceId        :
Version           : 14.0.27629.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x86__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x86__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 14.0.27629.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x64__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok
```

**Reply by DHowett** [reply]
Whoa!

Can you grab the latest version from [here (x64)](https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx) and install that, then try to install Terminal?

**Reply by melanchall** [reply]
Unfortunately I get the same error message, saying I'm already on newer version :( Now the command you've shown above outputs three records:

```
Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X86
ResourceId        :
Version           : 14.0.27629.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x86__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x86__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 14.0.27629.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x64__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 14.0.30035.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok
```

**Reply by DHowett** [reply]
Argh! Even that version is out of date compared to what Terminal needs!

Sorry to keep pulling you in random directions. Would you mind downloading the Windows Terminal 1.12.10732 *Preinstall Kit* (it's the 60+ MB zip file in the release section), and installing the x64 Microsoft.VCLibs package from inside it?

It looks like our distribution story here isn't 100% solid, so we'll have to revisit how we depend on the VC runtime. . . again.

**Reply by melanchall** [reply]
@DHowett Same error. Now PowerShell gives four records:

<details>
  <summary>PowerShell output</summary>

  ```
Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X86
ResourceId        :
Version           : 14.0.27629.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x86__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x86__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 14.0.27629.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.27629.0_x64__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 14.0.30035.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.30035.0_x64__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok

Name              : Microsoft.VCLibs.140.00.UWPDesktop
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US
Architecture      : X64
ResourceId        :
Version           : 14.0.30704.0
PackageFullName   : Microsoft.VCLibs.140.00.UWPDesktop_14.0.30704.0_x64__8wekyb3d8bbwe
InstallLocation   : C:\Program Files\WindowsApps\Microsoft.VCLibs.140.00.UWPDesktop_14.0.30704.0_x64__8wekyb3d8bbwe
IsFramework       : True
PackageFamilyName : Microsoft.VCLibs.140.00.UWPDesktop_8wekyb3d8bbwe
PublisherId       : 8wekyb3d8bbwe
IsResourcePackage : False
IsBundle          : False
IsDevelopmentMode : False
NonRemovable      : False
IsPartiallyStaged : False
SignatureKind     : Store
Status            : Ok
  ```

</details>

**Reply by DHowett** [reply]
I think we have a root cause; check out my last comment in #12816. I would link it but I'm mobile at the moment :)

Edit: here https://github.com/microsoft/terminal/issues/12816#issuecomment-1087739306

**Reply by DHowett** [reply]
ALRIGHT, Finally I can come back to you with a solution. I shared a link to the dism bundle in #12816 -- can you try installing the `msixbundle` from inside it? It's got a weird hashed name, but... eh.

**Reply by melanchall** [reply]
@DHowett Thanks. Now update available, BUT

![image](https://user-images.githubusercontent.com/26361185/162569441-0b857d1b-4dfd-40c1-b692-d7500d8552cf.png)

This progress is more than hour :( So installation stuck and don't move.

**Reply by MOHAMED-EHAB-DEV** [reply]
Please tell me how to install the app because microsoft store have a problem

**Reply by zadjii-msft** [reply]
@Programing-School Please read the README: https://github.com/microsoft/terminal#via-github

**Reply by zadjii-msft** [reply]
@melanchall @DHowett UHG I hate that that issue is back: https://github.com/microsoft/terminal/issues?q=is%3Aissue+%22installing+required+frameworks%22

You might want to use the Win10 preinstall kit, because that one bundles the framework packages straight up in our msix.

**Comment by DanRotaru** [comment]
When new win ui (like wt preview) ?

**Reply by DHowett** [reply]
Depends. We are trying to get this out to people, but we are worried about the impact it's going to have on our users. Do you want the new UI?

**Reply by DanRotaru** [reply]
yep

**Reply by DHowett** [reply]
1.12.1098 is rolling out with the Win11 UI

**Comment by MilioD** [comment]
> This release was made available to insiders in the Dev and Beta channels on March 25

I'm in the Windows Insiders Beta channel on OS build 22581.100, but I still have Terminal version 1.12.10393.0. I know it has only been 2 days and that I could get the update through `winget`. Also I use the Terminal Preview on the latest version already, so it isn't an issue for me. I just want to know if there is some kind of A/B testing going on with the versions because the same thing happened to me with the previous Terminal update which I only received on the 23<sup>rd</sup> of March.

**Reply by DHowett** [reply]
Oh, dang! I apparently only published this to Dev channel. Fixing. :smile:

**Reply by MilioD** [reply]
Sounds good! Thanks a lot for your help with this and all your work on the Terminal.

**Comment by DimitriDR** [comment]
Note: it is the **last installable version** on **Windows Server** (or machines that don't have MS Store) via `Add AppxPackage`.
